### ISSUE TITLE
Errors in labels

### ISSUE THREAD

**Issue by mattdistefano** [angular/angular.io#1147]
Been getting my feet wet with Angular 2 and noticed that a lot of the docs and examples are using labels that lack any programmatic association with their corresponding input, which is an accessibility fail (see https://www.w3.org/WAI/tutorials/forms/labels/ for correct usage).

For instance, the [Tour of Heros](https://angular.io/docs/ts/latest/tutorial/toh-pt1.html) does not use any association, while the [forms guide](https://angular.io/docs/ts/latest/guide/forms.html) includes a `for` attribute on the `label`s, but without the necessary `id` on the related `input`. The latter example will fail HTML5 validation too, and, IMO, makes the example confusing (I had to dig around elsewhere to see if maybe one of the other directives was automagically setting an `id` at runtime).

I totally get that the code samples are trying to focus on a single concept and can't address all possible concerns (introducing aria here wouldn't be appropriate, for instance). But I think this would be worth fixing - the current usage is arguably just wrong, and at best is an anti-pattern that promotes inaccessible coding.
